### PR TITLE
Pending list plugin, CLI & API

### PR DESCRIPTION
**Pull request by liiight**
### Motivation for changes:
Add an ability to manually approve or reject entries.

### Detailed changes:

- Created `pendling_list` list interface plugin.
- Added CLI ability
- Added API

### Config usage:
The plugin is a regular list interface plugin. Adding to the list is done as usual:
```yaml
rss: url
accept_all: yes
list_add:
  - pending_list: new_movies
```
All entries that are added to it are added with `approved = False`.
When used as an input, it'll produce only entries that have `approved == True`.
```yaml
pending_list: new_movies
accept_all: yes
list_add:
  - movie_list: movies
```
#### To Do:

- [x] API
- [x] API Tests
- [ ] Match approval operation to current one in `pending_approval`

### CODE REVIEW

File: flexget/plugins/cli/wait_list.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+    global_entry_parser = ArgumentParser(add_help=False)
+    global_entry_parser.add_argument('entry', help='Can be entry title or ID')
+
+    attributes_parser = ArgumentParser(add_help=False)
+    attributes_parser.add_argument('--attributes', metavar='<attributes>', nargs='+', type=attribute_type,
+                                   help='Can be a string or a list of string with the format imdb_id=XXX,'
+                                        ' tmdb_id=XXX, etc')
+    list_name_parser = ArgumentParser(add_help=False)
+    list_name_parser.add_argument('list_name', nargs='?', default='wait_entries',
```
Comment (paranoidi) [inline flexget/plugins/cli/wait_list.py]: default names for movies_list is "movies" and (hopefully) for series it's "series". This should be "waiting" to be in line?

File: flexget/plugins/cli/wait_list.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+            db_entry = get_wait_entry_by_id(entry_list.id, int(options.entry), session=session)
+        except NoResultFound:
+            console(
+                'Could not find matching wait entry with ID {} in list `{}`'.format(int(options.entry),
+                                                                                    options.list_name))
+            return
+        except ValueError:
+            db_entry = get_wait_entry_by_title(entry_list.id, options.entry, session=session)
+            if not db_entry:
+                console(
```
Comment (paranoidi) [inline flexget/plugins/cli/wait_list.py]: odd indentation

File: flexget/plugins/cli/wait_list.py [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    with Session() as session:
+        try:
+            entry_list = get_list_by_exact_name(options.list_name)
+        except NoResultFound:
+            console('Could not find wait list with name `{}`'.format(options.list_name))
+            return
+        try:
+            db_entry = get_wait_entry_by_id(entry_list.id, int(options.entry), session=session)
+        except NoResultFound:
+            console(
```
Comment (paranoidi) [inline flexget/plugins/cli/wait_list.py]: odd indentation

**Comment by liiight** [comment]
Closing, not a good idea :)

**Comment by liiight** [comment]
Reopened, it is a good idea :laughing:

**Comment by liiight** [comment]
Reopened as https://github.com/Flexget/Flexget/pull/1661

**Pull request closed without merging**